src/decoder/stateless/h264.rs

```diff
@@ -1093,6 +1093,18 @@ where
         timestamp: u64,
         slice: &Slice,
     ) -> Result<CurrentPicState<B::Picture>, DecodeError> {
+        // Start by securing the backend picture before modifying our state.
+        let first_field = self.codec.find_first_field(&slice.header)?;
+        let mut backend_pic = if let Some(first_field) = &first_field {
+            self.backend.new_field_picture(timestamp, &first_field.1)
+        } else {
+            self.backend.new_picture(timestamp)
+        }
+        .map_err(|e| match e {
+            StatelessBackendError::OutOfResources => DecodeError::NotEnoughOutputBuffers(1),
```

Comment:
Yes, we will definitely need more error variants. For instance V4L2 will fail if it cannot secure an **input** buffer, which is a different problem. I've preferred to not address this in this series since it's not directly related, and I still don't have a good idea of which variants we will need (will probably a better picture on hte next revision of V4L2 stateless).

I also don't like the `map_err` - hopefully we can have a single "resource unavailable" error type that `new_picture` will return, and `#[from]` that into a single `DecodeError` variant that the client can look into to understand which resource is missing. Still need to think whether this is the right design.

Reply:
> I've preferred to not address this in this series since it's not directly related

Introducing new variants or error types indeed isn't. But IMHO clarifying the effects of `StatelessBackendError::OutOfResources` on the client event, I think should happen sooner then later, especially with a new backend on the horizon. However I am leaving this up to :smile:

Reply:
Yeah, better to address this now indeed. I have added one extra commit to try and improve the situation, it doesn't get it 100% there but at least clarifies the type of resource missing and allowed me to remove that ugly `.map_err`.